packages/client/src/entities/user/model.ts

```diff
@@ -0,0 +1,37 @@
+import { createSlice } from '@reduxjs/toolkit'
+
+export type User = {
+  id: number
+  login: string
+}
+
+type InitialStateProps = User & {
+  isAuthenticated: boolean,
+}
+
+const initialState: InitialStateProps = {
+  id: 0,
+  login: '',
+  isAuthenticated: false
+}
+
+export const userModel = createSlice({
+  name: 'user',
+  initialState,
+  reducers: {
+    setUser: (state, { payload }) => {
+      // устанавливаем инфу о юзере в localStorage и в стейт
+      console.log({state, payload})
+    },
+
+    clearUser: () => {
+      // deleteUser()
+      // deleteToken()
+      // return initialState
+    }
+  }
+})
+
+export const { setUser, clearUser } = userModel.actions
+
+export default userModel.reducer
```

Comment:
Можно лучше: стоит также сразу типизировать хуки `useSelector` и `useDispatch`. К примеру так это делается в документации

https://react-redux.js.org/using-react-redux/usage-with-typescript#define-typed-hooks

```js
import type { TypedUseSelectorHook } from 'react-redux'
import { useDispatch, useSelector, useStore } from 'react-redux'

export type RootState = ReturnType<typeof store.getState>
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch
// Use throughout your app instead of plain `useDispatch` and `useSelector`
export const useAppDispatch: () => AppDispatch = useDispatch
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector

```